checker/main.go

```diff
@@ -0,0 +1,207 @@
+// Copyright 2020 Red Hat, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/RedHatInsights/insights-content-service/groups"
+	"github.com/RedHatInsights/insights-results-aggregator/content"
+	"github.com/RedHatInsights/insights-results-aggregator/logger"
+	"github.com/rs/zerolog/log"
+)
+
+// groupConfigMap is a shorthand for the map used to store the group configuration.
+type groupConfigMap map[string]groups.Group
+
+var (
+	groupConfigPath string = "./groups_config.yaml"
+	contentDirPath  string = "./content/"
+)
+
+func init() {
+	flag.StringVar(&groupConfigPath, "config", groupConfigPath, "Path to the group configuration YAML file.")
+	flag.StringVar(&contentDirPath, "content", contentDirPath, "Path to the content directory (the one containing the 'config.yaml' file).")
+	flag.Parse()
+}
```

Comment:
Apparently this method is not directly invoked. Is it some kind of "Golang magic"?

Reply:
yes, it is called when the module is initialized

Reply:
https://medium.com/golangspec/init-functions-in-go-eac191b3860a